### DISCUSSION TITLE
How can I make `make:entity`'s generated entity class extend a base entity class?

### DISCUSSION THREAD

**Discussion by jerrac** [symfony/symfony#48017] [Q&A]
My initial thought was to use a custom `Entity.tpl.php` file for `make:entity`. To do that, I thought to create my own Maker command. 

So I started that by copying `< MakerBundle >/src/Maker/MakeEntity.php` and trying to adjust it to fit my needs. When I hit the point of needing to copy `< MakerBundle >/src/Generator.php` because of how it uses a relative path to the `.tpl.php` file, I thought I had better stop and ask for advice. I don't think I should have to copy nearly every Entity related file in the bundle to accomplish my goal...

All I really need to do is add `extends App\Entity\FieldBase` to the `class` line, and remove the `id` stub.

What is the best method to make `make:entity`'s generated entity class extend a base entity class? 

Thanks in advance!

**Comment by stof** [comment]
This is not a feature supported in MakerBundle for now.

**Reply by jerrac** [reply]
Er, that's why I was trying to create a new Maker class. I'm just looking for guidance on how best to implement what I need without over-complicating things (which is what having to copy all that code seems to to be...).

**Comment by ghost** [comment]
Why can't you add this `extends SomeBaseClass` by hand? This is 3 seconds work.

**Reply by jerrac** [reply]
It's needed because of how I want my app to work for people other than me.
